### PR TITLE
Script to regen test fixture

### PR DESCRIPTION
**Pull request by gabelevi**
I really super extra hate this test suite. It's fine for adding a few tests, but when you need to update the AST for tens of tests it really sucks. This script runs all the tests and pretty prints their results back into the test fixture.

There are some downsides to this script:
1) Damages git blame
2) Nukes comments (luckily there's only one)

There are some upsides to this script:
1) Makes me not want to kill myself
2) Demonstrates how this test suite is a pain to work with.
3) Enables changes that were unthinkable before due to the pain of updating the tests, like fixing some property from undefined to null.

I've included the results of this script. I think this script might be useful to other people. I can always not check in the output of the script, and instead try and just merge the updated tests that I care about.

### CODE REVIEW

**Comment by gabelevi** [comment]
BTW I tried the following:

1) Update fbtests.js using this script
2) Make a change to esprima.js
3) Regenerate fbtests.js using this script
4) Rebase without the changes from step (1)

This led to the worst merge conflict that I've ever seen. Possibly due to the huge change in the line numbers (script gen'd file is 1.5k lines shorter) and the commonly occurring patterns.

File: package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
         "lint": "node tools/check-version.js && node node_modules/eslint/bin/eslint.js esprima.js && node node_modules/jslint/bin/jslint.js esprima.js",
         "coverage": "npm run-script analyze-coverage && npm run-script check-coverage",
         "analyze-coverage": "node node_modules/istanbul/lib/cli.js cover test/runner.js",
-        "check-coverage": "node node_modules/istanbul/lib/cli.js check-coverage --statement -11 --branch -28 --function 99.69",
+        "check-coverage": "node node_modules/istanbul/lib/cli.js check-coverage --statement -11 --branch -30 --function 99.69",
```
Comment (jeffmo) [inline package.json:57]: necessary? Would prefer to leave upstream stuff like this as-is (unless there's a really good reason?)

File: tools/generate-whole-fixture.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    .replace(/[\f]/g, '\\f')
+    .replace(/[\n]/g, '\\n')
+    .replace(/[\r]/g, '\\r')
+    .replace(/[\t]/g, '\\t')
+    .replace(/[']/g, "\\'");
+}
+
+var ast, result;
+for (section in testFixture) {
+  out += "    '"+section+"': {\n";
```
Comment (jeffmo) [inline tools/generate-whole-fixture.js]: `JSON.stringify(section)` (escapes stuff like quotes/etc properly)

File: tools/generate-whole-fixture.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    comment: false,
+    range: true,
+    loc: true,
+    tokens: false,
+    raw: true,
+    tolerant: false
+};
+
+function escape_content(content) {
+  return content
```
Comment (jeffmo) [inline tools/generate-whole-fixture.js]: I think this should take care of things for you (in a perhaps more comprehensive way), right?

`return JSON.stringify(content).replace(/\\/g, '\\\\')`

**Comment by gabelevi** [comment]
By the way, https://github.com/gabelevi/esprima/commit/c3ff2d251b0420340801ef93a98de58ef69b0fce is an example of a change that becomes really easy with this script

File: tools/generate-whole-fixture.js (hunk 2 above)
Comment (RReverser) [inline tools/generate-whole-fixture.js]: Or even try [serialize-js](https://www.npmjs.org/package/serialize-js) - I created it for myself exactly for same purposes (generating AST tests). The only difference from simple JSON is that it doesn't quote key names when it's not 100% necessary + doesn't add indentation on single-line things like `key: [**here**1**and here**]`.

**Comment by gabelevi** [comment]
I've opened an issue upstream to gauge how people feel about switching to an autogenerated AST rather than a handcoded one. https://code.google.com/p/esprima/issues/detail?id=606&start=100

File: esprima.js [hunk 4]
```diff
@@ -3906,6 +3905,7 @@ parseYieldExpression: true, parseAwaitExpression: true
                 }
                 indexer = parseObjectTypeIndexer();
             } else {
+                marker = markerCreate();
```
Comment (gabelevi) [inline esprima.js:3908]: This was a small bug fix uncovered by the updated test hardness.

**Comment by gabelevi** [comment]
Ok, so here's what's new:

1) Split out `stringify()` into it's own module
2) Separated fbtest.js into fbtest.js and fbtest.rec.js. The former contains the test strings and the latter contains the recorded asts/errors
3) The test runner runs fbtest.rec.js now

**Pull request merged**